.travis.yml

```diff
@@ -1,15 +1,25 @@
-# Set up the Rust toolchain.
+---
+env: 
+  - FORMATTING=true
+  - FORMATTING=false
 language: rust
-rust:
+matrix: 
+  allow_failures: 
+      - env: "FORMATTING=false"
+        rust: nightly
+  exclude: 
+      - env: "FORMATTING=true"
+        rust: stable
+      - env: "FORMATTING=true"
+        rust: beta
+      - env: "FORMATTING=true"
+        os: osx
+        rust: nightly
```

Comment:
this looks rather unclear to me. Is the intention only test the formatting on Linux-nightly? If so, wouldn't it be simpler to just have an "if Linux && nightly { test formatting }" as opposed to encoding that into the build matrix?

Reply:
That is the intention, but the reason it was done this way is because standard Linux nightly is an allowed failure, we don't want formatting failures to be an allowed failure, so a dedicated machine is added for it that is not an allowed failure.

Reply:
The intention is to create another build machine. Providing two versions of one env var multiplies them, so we have to exclude all the duplicates except one.

Reply:
:/ not sure if the confusing config is worth an extra machine here. rustfmt check is supposedly light enough to be done together with some building job

Reply:
That was the original intent, but here's the constraints I've been given:

- We must use rustfmt-nightly, because rustfmt isn't maintained
- Formatting failure must fail the build, otherwise this does us no good
- Failing to build on nightly must not fail the build, as nightly can fail for reasons unrelated to the PR in question

With those constraints this was the solution I came up with.

Reply:
Ok, I understand. Is it safe to assume that nightly failures are infrequent? In this case, I'd prefer the rustfmt checks to be failable as well. I don't see a point in strict enforcement of formatting, since failure to format doesn't block anyone from working/testing the product. Hopefully, people just keep an eye on nightly failures and react accordingly. Common sense can save quite a bit of time/nerves from PRs rejected purely on the formatting basis.

Reply:
Then you disagree with the premise of the pull request, which is that formatting of the code base should be enforced.  A constantly formatted code base won't see changes when people run formatting tools on their code, so people can both easily submit well formatted code and not incur additional reviewer work in sorting out the "just formatting" changes.  That's my motivation here.